File: Codes/MaxSubarray.cpp
class Solution {
public:
    int maxSubArray(int A[], int n) {
        int sum, tempSum;
        sum = tempSum = A[0];
        for (int i = 1; i < n; i++) {
            if (tempSum < 0) {
                tempSum = A[i];
            } else {
                tempSum += A[i];
            }
            if (sum <= tempSum) sum = tempSum;
        }

        return sum;
    }
};

File: Codes/MaxSubarray.py
class Solution:
    # @param A, a list of integers
    # @return an integer
    def maxSubArray(self, A):
        if len(A) == 0:
            return 0
        maxSofar, maxAll = A[0], A[0]
        for x in A[1:]:
            if maxSofar < 0:
                maxSofar = x
            else:
                maxSofar += x
            if maxSofar > maxAll:
                maxAll = maxSofar
        return maxAll